core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -256,19 +251,41 @@ class ZkUtils(val zkClient: ZkClient,
   }
 
   /**
-   * Register brokers with v2 json format (which includes multiple endpoints).
+   * Register brokers with v3 json format (which includes multiple endpoints and rack).
```

Comment:
I will update the document. I think it is safer not to register the rack when version <  3 and having rack in v2 violates the specification anyway which might be confusing to users.